assets/app/view/game/round/stock.rb

```diff
@@ -103,28 +103,28 @@ def render_ipoed
             if @step.can_buy?(@current_entity, pool_share)
               children << h(:button, { on: { click: buy_pool } }, 'Buy Market Share')
             end
+          end
 
-            # Allow privates to be exchanged for shares
-            @game.companies.each do |company|
-              company.abilities(:exchange) do |ability|
-                next unless ability.corporation == @selected_corporation.name
-                next unless company.owner == @current_entity
+          # Allow privates to be exchanged for shares
```

Comment:
Very long methods...

Personally I do like extract-method so I would consider a local allow_privates_to_be_exchanged_for_shares method, but that's me.

Reply:
Same here, but it'd be a much bigger rewrite to isolate that block

Reply:
Same here, but unfortunatly it relies on a lot of variables local to that method.